### ISSUE TITLE
autoInject doesn't unpack results into the callback

### ISSUE THREAD

**Issue by steverobb** [caolan/async#1099]
The original version of autoInject unpacked arguments into the final callback.  The current version does not do this:

```
async.autoInject({
    a: function(cb) { cb(null, 1); },
    b: function(cb) { cb(null, 2); }
}, function(err, a, b) {
    console.log('a: ' + a + ' b: ' + b);
});
```

Expected: a: 1, b: 2
Actual: a: [object Object], b: undefined

The documentation still refers to the original behaviour.

**Comment by steverobb** [comment]
To be clear, I see this as an oversight made during the porting of the code, not a documentation error.

**Comment by aearly** [comment]
Closed by #1100
